File: custom/qt-ui-framework/assets/pyqt6/mainwindow.py
"""
PyQt6 Main Window
"""

from PyQt6.QtWidgets import QMainWindow, QWidget, QVBoxLayout, QLabel

class MainWindow(QMainWindow):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.SetupUI()
        self.ConnectSignals()
    
    def SetupUI(self):
        central_widget = QWidget(self)
        self.setCentralWidget(central_widget)

        layout = QVBoxLayout(central_widget)

        label = QLabel("Hello, PyQt6!", self)
        layout.addWidget(label)

        self.setWindowTitle("My Application")
        self.resize(800, 600)

    def ConnectSignals(self):
        # Connect signals and slots here.
        # Example: button.clicked.connect(self.OnButtonClicked)
        pass

File: custom/qt-ui-framework/assets/cpp-widgets/mainwindow.cpp
#include "mainwindow.h"
#include <QWidget>
#include <QVBoxLayout>
#include <QLabel>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
{
    SetupUI();
    ConnectSignals();
}

MainWindow::~MainWindow()
{
}

void MainWindow::SetupUI()
{
    QWidget *central_widget = new QWidget(this);
    setCentralWidget(central_widget);

    QVBoxLayout *layout = new QVBoxLayout(central_widget);

    QLabel *label = new QLabel("Hello, Qt!", this);
    layout->addWidget(label);

    setWindowTitle("My Application");
    resize(800, 600);
}

void MainWindow::ConnectSignals()
{
    // Connect signals and slots here.
    // Example: connect(button, &QPushButton::clicked, this, &MainWindow::OnButtonClicked);
}
